pkg/controller/backup_controller.go

```diff
@@ -488,6 +470,16 @@ func getObjectStoreForLocation(location *api.BackupStorageLocation, manager plug
 		return nil, err
 	}
 
+	// add the bucket name to the config map so that object stores can use
+	// it when initializing. The AWS object store uses this to determine the
+	// bucket's region when setting up its client.
+	if location.Spec.ObjectStorage != nil {
+		if location.Spec.Config == nil {
+			location.Spec.Config = make(map[string]string)
+		}
+		location.Spec.Config["bucket"] = location.Spec.ObjectStorage.Bucket
```

Comment:
`location.Spec.Config` will contain whatever config key-value pairs the user provides when setting up a backup location, so it might have data in it already.  `bucket` is not a user-specified value for that config, so this would be the only place it's being inserted.  But, when we add `bucket`, we also want to retain any other user-specified values in the map, if it exists

Reply:
Also note that this is not being saved to the API, it's just an in-proc thing.

Reply:
Thank you for the explanation, so helpful.

Reply:
@carlisia @nrb good to merge? I'm holding off on testing until this goes in